internal/threads/repository/thread_repo.go

```diff
@@ -0,0 +1,109 @@
+package repository
+
+import (
+	"github.com/jmoiron/sqlx"
+	. "github.com/yletamitlu/tech-db/internal/helpers"
+	"github.com/yletamitlu/tech-db/internal/models"
+	"github.com/yletamitlu/tech-db/internal/threads"
+)
+
+type ThreadPgRepos struct {
+	conn *sqlx.DB
+}
+
+func NewThreadRepository(conn *sqlx.DB) threads.ThreadRepository {
+	return &ThreadPgRepos{
+		conn: conn,
+	}
+}
+
+func (tr *ThreadPgRepos) SelectBySlug(slug string) (*models.Thread, error) {
+	thread := &models.Thread{}
+
+	if err := tr.conn.Get(thread,
+		`SELECT * from threads where slug = $1`,
+		slug); err != nil {
+		return nil, PgxErrToCustom(err)
+	}
+
+	if thread.Slug == "" {
+		return nil, nil
+	}
+
+	return thread, nil
+}
+
+func (tr *ThreadPgRepos) SelectById(id int) (*models.Thread, error) {
+	thread := &models.Thread{}
+
+	if err := tr.conn.Get(thread,
+		`SELECT * from threads where id = $1`,
+		id); err != nil {
+		return nil, PgxErrToCustom(err)
+	}
+
+	return thread, nil
+}
+
+func (tr *ThreadPgRepos) SelectByForumSlug(slug string, limit int, desc bool, since string) ([]*models.Thread, error) {
+	var threads []*models.Thread
+
+	if since == "" {
+		if desc {
+			if err := tr.conn.Select(&threads,
+				`SELECT * from threads where forum_slug = $1 order by created_at desc limit $2`,
+				slug, limit); err != nil {
+				return nil, PgxErrToCustom(err)
+			}
+		} else {
+			if err := tr.conn.Select(&threads,
+				`SELECT * from threads where forum_slug = $1 order by created_at limit $2`,
+				slug, limit); err != nil {
+				return nil, PgxErrToCustom(err)
+			}
+		}
+	} else {
+		if desc {
+			if err := tr.conn.Select(&threads,
+				`SELECT * from threads where forum_slug = $1 and created_at <= $2 order by created_at desc limit $3`,
+				slug, since, limit); err != nil {
+				return nil, PgxErrToCustom(err)
+			}
+		} else {
+			if err := tr.conn.Select(&threads,
+				`SELECT * from threads where forum_slug = $1 and created_at >= $2 order by created_at limit $3`,
+				slug, since, limit); err != nil {
+				return nil, PgxErrToCustom(err)
+			}
+		}
+	}
```

Comment:
Это все нужно переосмыслить, разбить на функции и облегчить
Тут повторяется по сути код, подумай, как его в зависимости от пришедших параметров разбить на функции и удобно использовать (в зависимости от desc true/false, since)